File: javapatterns/src/main/java/org/iut/nantes/facade/avion/Facade.java
package org.iut.nantes.facade.avion;

public class Facade {
    public void decollage() {
        GestionElectrique gestionElectrique = new GestionElectrique();
        GestionEssence gestionEssence = new GestionEssence();
        GestionMecanique gestionMecanique = new GestionMecanique();

        gestionElectrique.batterieOn();
        gestionElectrique.chauffeCarbu();

        gestionEssence.autonomie();
        gestionEssence.mixtureRiche();
        gestionEssence.injection();

        gestionMecanique.debattement();
        gestionMecanique.voletOn();
        gestionMecanique.verriereOpen();

        gestionMecanique.demarrage();
    }

    public void atterrissage() {
        GestionElectrique gestionElectrique = new GestionElectrique();
        GestionEssence gestionEssence = new GestionEssence();
        GestionMecanique gestionMecanique = new GestionMecanique();

        gestionElectrique.batterieOff();
        gestionMecanique.freinOff();
        gestionElectrique.chauffeCarbuOff();
        gestionEssence.mixturePauvre();
    }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/facade/avion/Facade.kt
package org.iut.nantes.facade.avion

class Facade {
    fun decollage() {
        val gestionElectrique = GestionElectrique()
        val gestionEssence = GestionEssence()
        val gestionMecanique = GestionMecanique()
        gestionElectrique.batterieOn()
        gestionElectrique.chauffeCarbu()
        gestionEssence.autonomie()
        gestionEssence.mixtureRiche()
        gestionEssence.injection()
        gestionMecanique.debattement()
        gestionMecanique.voletOn()
        gestionMecanique.verriereOpen()
        gestionMecanique.demarrage()
    }

    fun atterrissage() {
        val gestionElectrique = GestionElectrique()
        val gestionEssence = GestionEssence()
        val gestionMecanique = GestionMecanique()
        gestionElectrique.batterieOff()
        gestionMecanique.freinOff()
        gestionElectrique.chauffeCarbuOff()
        gestionEssence.mixturePauvre()
    }
}